test/unit/kubernetes-deploy/kubernetes_resource/daemon_set_test.rb

```diff
@@ -38,22 +40,78 @@ def test_deploy_passes_when_updated_available_does_match
       "observedGeneration": 2,
     }
 
-    ds_template = build_ds_template(status: status)
-    ds = build_synced_ds(template: ds_template)
+    ds_template = build_ds_template(filename: 'daemon_set.yml', status: status)
+    ds = build_synced_ds(ds_template: ds_template)
+    assert_predicate(ds, :deploy_succeeded?)
+  end
+
+  def test_deploy_passes_when_ready_pods_for_one_node
+    status = {
+      "desiredNumberScheduled": 1,
+      "updatedNumberScheduled": 1,
+    }
+    ds_template = build_ds_template(filename: 'daemon_set.yml', uid: OWNER_UID, generation: 2, status: status)
+    pod_template = build_pod_template(filename: 'daemon_set_pod.yml')
+    node_template = build_node_template(filename: 'node.yml')
+    ds = build_synced_ds(ds_template: ds_template, pod_templates: pod_template, node_templates: node_template)
+    assert_predicate(ds, :deploy_succeeded?)
+  end
+
+  def test_deploy_passes_when_ready_pods_for_multiple_nodes
+    status = {
+      "desiredNumberScheduled": 3,
+      "updatedNumberScheduled": 3,
+      "numberReady": 2,
+    }
+    ds_template = build_ds_template(filename: 'daemon_set.yml', uid: OWNER_UID, generation: 2, status: status)
+    pod_template = build_pod_template(filename: 'daemon_set_pod.yml')
+    node_template = build_node_template(filename: 'node.yml')
```

Comment:
3 desired, 2 ready and 1 node seems like a weird state that would never be observed on a single sync cycle--you'd for sure see three nodes given that status. If we go with the node list caching approach, this test will need to multiple sync cycles. Otherwise, it will need to use multiple nodes with manipulated creation timestamps.

Reply:
I agree these states are weird. the intent was to simulate where a node has been added by autoscaling. I'm going to adjust the numbers accordingly.

> If we go with the node list caching approach, this test will need to multiple sync cycles

I can add a separate test that handles the multiple-sync case

Reply:
I think that `test_deploy_passes_when_ready_pods_but_node_added` should cover this, but i wasn't sure how I could use the same daemonset instance, though I'm not totally sure it matters.

Can you think of a way to reuse the daemonset instance and update the status transparently? Ultimately to test logic around cache I think that's needed.